### PR TITLE
oast: Interactsh Bugfix: ServerUrl change in Options

### PR DESCRIPTION
**Pull request by denniskniep**
OAST Interactsh: ServerUrl change in Options deregisters old ServerUrl and registers new ServerUrl

Signed-off-by: Dennis Kniep <kniepdennis@gmail.com>

### CODE REVIEW

File: addOns/oast/CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
-
+### Fixed
+- Interactsh: ServerUrl change in Options deregisters old ServerUrl and registers new ServerUrl
```
Comment (thc202) [inline addOns/oast/CHANGELOG.md:10]: `server URL`

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java [hunk 2]
```diff
@@ -131,18 +133,46 @@ public void sessionChanged() {
...
 
     @Override
     public void optionsChanged(OptionsParam optionsParam) {
-        if (currentPollingFrequency != param.getPollingFrequency()) {
-            stopPoller();
-            startService();
-            currentPollingFrequency = param.getPollingFrequency();
-            LOGGER.debug(
-                    "Updated Interactsh Polling frequency to {} seconds.", currentPollingFrequency);
+        if (!StringUtils.equals(currentServerUrl, param.getServerUrl())) {
```
Comment (thc202) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java:142]: `Objects.equals`

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java [hunk 3]
```diff
@@ -131,18 +133,46 @@ public void sessionChanged() {
...
+            try {
+                stopPoller();
+                deregister();
```
Comment (thc202) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java:145]: `stopPoller` could be moved to `deregister`, less error prone (that's duplicated in other place).

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java [hunk 4]
```diff
@@ -131,18 +133,46 @@ public void sessionChanged() {
...
+                register();
+                LOGGER.debug(
+                        "Updated Interactsh params - ServerUrl:{} and polling frequency: {} seconds.",
```
Comment (thc202) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java:148]: `server URL` and following.

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java [hunk 5]
```diff
@@ -125,24 +127,50 @@ public void stopService() {
...
-            startService();
-            currentPollingFrequency = param.getPollingFrequency();
-            LOGGER.debug(
-                    "Updated Interactsh Polling frequency to {} seconds.", currentPollingFrequency);
+        if (!Objects.equals(currentServerUrl, param.getServerUrl())) {
+            try {
+                deregister();
+                register();
+                LOGGER.debug(
+                        "Updated Interactsh params - server URL:{} and polling frequency: {} seconds.",
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java:146]: URL colon space to be consistent with others?

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**